document new date queries

bendyorke opened the pull request:
Add documentation for relative date queries. See screenshot below.

1. Documentated deprecated queries and their respective upgrade path
2. Updated examples with new query syntax
3. Provided examples for all abstract representations, however defaulted to collapsed for improved readability. Demarcated with an _(examples within)_

<img width="1085" alt="Screenshot 2023-02-08 at 13 50 29" src="https://user-images.githubusercontent.com/3488879/217534430-643faad1-f7b3-44cc-94b0-ee45c681c1be.png">

bendyorke committed:
document new date queries

logseq-cldwalker commented on pages/Advanced Queries.md:
```diff
@@ -39,22 +39,87 @@ description:: Advanced queries are written with [Datalog](https://en.wikipedia.o
+		- #### Deprecated inputs
+		  collapsed:: true
+			- Previously we supported different inputs between basic queries and advanced queries. These have now been unified, however we still support the follow queries for the time being. If you are using any of the following, plus upgrade them to the newer syntax mentioned for each item
+			- `:Xd` - Date for X number of days after today. **New syntax:** `:+Xd`
```
Confusingly enough this used to match on -7d and +7d but it always short circuited on -7d. Glad we have the new syntax. I can fix this and related examples below

logseq-cldwalker commented on pages/Advanced Queries.md:
```diff
@@ -39,22 +39,87 @@ description:: Advanced queries are written with [Datalog](https://en.wikipedia.o
 			- `:current-block` - `:db/id` of current block. See [this example](((63bc5e11-24f1-45fd-945d-4a272e5ecf0d)))
 			- `:parent-block` - `:db/id` of parent block
-		- Inputs for querying journal page blocks and rules like `between`
+		- #### Special date inputs (useful for querying journal pages or blocks relative to today's date)
```
Headers have permanent links based on the first line of content so it'd be good to move descriptions down a line to have a readable and more stable link. See https://docs.logseq.com/#/page/publishing/block/available%20urls for more. I can fix this

logseq-cldwalker commented on pages/Advanced Queries.md:
```diff
@@ -39,22 +39,87 @@ description:: Advanced queries are written with [Datalog](https://en.wikipedia.o
+				- `:today-HHMM` - returns a timestamp for HH:MM:00.000 today
+				- `:today-HHMMSS` - returns a timestamp for HH:MM:SS.000 today
+				- `:today-HHMMSSmmm` - returns a timestamp for HH:MM:SS.mmm today
+		- #### Deprecated inputs
```
Glad you have this section to ease the transition

logseq-cldwalker commented on pages/Advanced Queries.md:
```diff
@@ -39,22 +39,87 @@ description:: Advanced queries are written with [Datalog](https://en.wikipedia.o
+			- `:+Xy` - Date for X number of years after today
+		- #### Relative timestamp inputs (useful for querying for block attributes that use millisecond timestamps e.g. `:block/created-at`)
+			- Relative timestamp inputs can be created by adding a **time suffix** to any of the relative date inputs. Multiple **time suffixes** cannot be combined.  The **time suffixes** that are supported are as follows:
+				- `-start` - returns the start of the day you are querying for, returning the timestamp for 00:00:00.000.  *(examples within)*
```
Great to see collapsed examples for these ❤️

logseq-cldwalker commented on pages/Advanced Queries.md:
```diff
@@ -167,7 +232,7 @@ description:: Advanced queries are written with [Datalog](https://en.wikipedia.o
 	            (task ?b #{"NOW" "DOING"})
 	            (between ?b ?start ?today)]
-	    :inputs [:14d :today]
+	    :inputs [:+14d :today]
```
See block description to see some of these should be -Xd. Fixing

logseq-cldwalker commented on pages/Advanced Queries.md:
```diff
@@ -229,7 +294,7 @@ description:: Advanced queries are written with [Datalog](https://en.wikipedia.o
 	            (task ?b #{"NOW" "LATER" "TODO" "DOING"})
 	            (between ?b ?start ?today)]
-	    :inputs [:56d :8d]
+	    :inputs [:+56d :+8d]
```
Chance to showcase the new syntax :)

logseq-cldwalker reviewed (approved):
@bendyorke Thanks for the most excellent docs! Users will be grateful for this 👍 🕺 Fixing up some minor things and merging

logseq-cldwalker committed:
Address things caught in review

- Give headers more readable and stable links
- Fix missing apostrophes and whitespace
- Fix examples
